### DISCUSSION TITLE
What's the deal with the branch being refs/heads/master in Cypress dashboard?

### DISCUSSION THREAD

**Discussion by tnrich** [cypress-io/cypress#18767] [Cypress Cloud]
![image](https://user-images.githubusercontent.com/2730609/140013393-83c54811-388d-4fa8-a61e-c40c9a609beb.png)

I'm trying to make a README badge right now that tracks the tests on the master branch: 
![image](https://user-images.githubusercontent.com/2730609/140013516-52d04bc9-dcfc-4001-8e6f-eb1538b84cbf.png)

Unfortunately, there seems to be a difference between master and refs/heads/master .. Not sure what that difference is though and why this is happening. Any help would be very appreciated!

Thanks,
Thomas

**Comment by tnrich** [comment]
Also, please note that putting `refs/heads/master` as the branch causes the badge to break: 
![image](https://user-images.githubusercontent.com/2730609/140013956-24527b1f-b505-45cc-a33d-dce60bfac7b6.png)

**Comment by tnrich** [comment]
@bahmutov do you have any ideas on why my runs are all being prefixed with the refs/heads or refs/pull 

![image](https://user-images.githubusercontent.com/2730609/141373199-f6fa21d7-5931-42bf-8bf7-ed1f791f82ca.png)

Not quite sure why that is the case. Here's my organization link: https://dashboard.cypress.io/organizations/c0e49af4-c50f-4aff-ad08-9bc0dbc934d5/projects

Thanks in advance 🙏

**Comment by bahmutov** [comment]
That’s how git and environment variables show the current branch

**Comment by tnrich** [comment]
@bahmutov any idea on how to get the badge to work with the `refs/heads/master` branch? Also why are some of the runs said to be against just `master` instead of `refs/heads/master`?

Thanks!

**Comment by bahmutov** [comment]
Honestly I don’t remember - there is commit-info module I believe that gets this info from environment variables but you would need to find this in code to see the full picture

**Comment by tnrich** [comment]
@bahmutov could you point me in the right direction for this code?

**Comment by bahmutov** [comment]
I would search for git commit words

**Reply by tnrich** [reply]
What words :) ?

**Comment by bahmutov** [comment]
“git commit”

**Comment by tnrich** [comment]
@bahmutov I've made a screen recording of the issue I'm seeing. Any advice for how to get the badge to JustWork™ would be awesome! Thank you

https://user-images.githubusercontent.com/2730609/141825505-ef839560-03df-4783-8355-cf1b5d081d3d.mov

!

**Comment by bahmutov** [comment]
Sorry Thomas, I don’t think I have time to help with this
